src/write.jl

```diff
@@ -12,7 +12,7 @@ function _variable_info(model::Model{T}, x) where {T}
     is_bool = (T == Bool) || MOI.is_valid(model, ci_zero)
     # Integer
     ci_int = MOI.ConstraintIndex{F,MOI.Integer}(x.value)
-    is_int = MOI.is_valid(model, ci_int)
+    is_int = MOI.is_valid(model, ci_int) || T <: Integer
```

Comment:
@odow I had to add this to make bridges work for `test_nqueens_alldiff`. It seems reasonable - if the model type is Int, then the variables should be integer (or boolean). What happened without it is that in `test_nqueens_alldiff`, when bridging a VAF in AllDifferent constraint to affine constraints and a VV in AllDifferent, the auxiliary variables were just Floats, which threw an error in MiniZinc because there can't be Float variables in an AllDifferent constraint.

Reply:
But it does come back to the question I have asked before: what is the meaning of `T` in `Model{T}`. In this case I am conveniently interpreting it partly as determining the types of variables.

Reply:
Yeah in JuMP we are conflating `T` with the coefficient type and the value type. Let's talk tomorrow?